diff -uNr busybox-1.29.1/include/libbb.h busybox-1.29.1.mod/include/libbb.h
--- busybox-1.29.1/include/libbb.h	2018-07-02 14:23:06.000000000 +0300
+++ busybox-1.29.1.mod/include/libbb.h	2018-07-31 14:48:08.164173520 +0300
@@ -2061,12 +2061,12 @@
  * use bb_default_login_shell and following defines.
  * If you change LIBBB_DEFAULT_LOGIN_SHELL,
  * don't forget to change increment constant. */
-#define LIBBB_DEFAULT_LOGIN_SHELL  "-/bin/sh"
+#define LIBBB_DEFAULT_LOGIN_SHELL  "-/system/bin/sh"
 extern const char bb_default_login_shell[] ALIGN1;
 /* "/bin/sh" */
 #define DEFAULT_SHELL              (bb_default_login_shell+1)
 /* "sh" */
-#define DEFAULT_SHELL_SHORT_NAME   (bb_default_login_shell+6)
+#define DEFAULT_SHELL_SHORT_NAME   (bb_default_login_shell+13)
 
 /* The following devices are the same on all systems.  */
 #define CURRENT_TTY "/dev/tty"
